libbeat/common/file/rotator.go

```diff
@@ -441,3 +444,39 @@ func (r *Rotator) rotateBySize(reason rotateReason) error {
 	}
 	return nil
 }
+
+func (r *Rotator) purgeOldIntervalOnStartup() error {
+	files, err := filepath.Glob(filepath.Dir(r.filename) + "/*.[0-9]")
+	if err != nil {
+		return errors.New("failed to list old interval logs during initialization")
+	}
+
+	if len(files) > 0 {
+		for index, fileName := range files {
+
+			fi, err := os.Stat(fileName)
+			if err != nil {
+				return errors.New("failed to access old interval logs during initialization")
+			}
+
+			logPrefix := r.intervalRotator.LogPrefix(r.filename, fi.ModTime())
+
+			err = os.Link(fileName, logPrefix)
+			if os.IsExist(err) {
+				err = os.Link(fileName, logPrefix+strconv.Itoa(index))
+			}
+			if os.IsPermission(err) {
+
```

Comment:
???

Reply:
also we have a method called safeRename.
you can rename, don't need to create links. also it makes it a bit difficult to properly rotate older files based on timestamp. 
whatever rotator strategy is used it should end up with `filename-{timestamp}` file and code for the `filename.N` should disappear (except the migration[rename] on startup)

Reply:
safeRename? can you tell me where it is? I did not found it. 
The link prevents writing over another file, rename does not offer that. 

Until now no major changes on the behavior was made. 
To change the rotation by size filenames will break all tests and will need more attention. 
Can we do this more carefully in a new Pull Request?